test/token/FTCVesting.spec.ts

```diff
@@ -0,0 +1,260 @@
+    let subjectVestingAmount: BigNumber;
+    let subjectVestingStart: BigNumber;
+    let subjectVestingCliff: BigNumber;
+    let subjectVestingEnd: BigNumber;
+    let subjectFtcVesting: FTCVesting;
+
+    beforeEach(async () => {
+      const now = await getLastBlockTimestamp();
+      subjectVestingStart = now.sub(60 * 60 * 24 * 185);
+      subjectVestingCliff = now;
+      subjectVestingEnd = now.add(60 * 60 * 24 * 546);
+      subjectVestingAmount = ether(1);
+
+      subjectFtcVesting = await deployer.token.deployFtcVesting(
+        index.address,
+        recipient.address,
+        treasury.address,
+        subjectVestingAmount,
+        subjectVestingStart,
+        subjectVestingCliff,
+        subjectVestingEnd,
+      );
+
+      await index.transfer(subjectFtcVesting.address, ether(1000));
+    });
+
+    async function subject(): Promise<ContractTransaction> {
+      return await subjectFtcVesting.connect(recipient.wallet).claim();
+    }
+
+    it("should make a claim", async () => {
+      const unclaimedAmount = await index.connect(recipient.wallet).balanceOf(recipient.address);
+      const preClaimBalance = await index.balanceOf(subjectFtcVesting.address);
+      await subject();
+      const claimedAmount = await index.connect(recipient.wallet).balanceOf(recipient.address);
+      const postClaimBalance = await index.balanceOf(subjectFtcVesting.address);
+
+      const hasClaimed = claimedAmount.gt(unclaimedAmount);
+      const hasLowerBalance = preClaimBalance.gt(postClaimBalance);
+
+      expect(hasClaimed).to.be.true;
+      expect(hasLowerBalance).to.be.true;
+    });
+  });
+
+  describe("#clawback", async () => {
+    let subjectVestingAmount: BigNumber;
+    let subjectVestingStart: BigNumber;
+    let subjectVestingCliff: BigNumber;
+    let subjectVestingEnd: BigNumber;
+    let subjectFtcVesting: FTCVesting;
+
+    beforeEach(async () => {
+      const now = await getLastBlockTimestamp();
+      subjectVestingStart = now.sub(60 * 60 * 24 * 185);
+      subjectVestingCliff = now;
+      subjectVestingEnd = now.add(60 * 60 * 24 * 546);
+      subjectVestingAmount = ether(1);
+
+      subjectFtcVesting = await deployer.token.deployFtcVesting(
+        index.address,
+        recipient.address,
+        treasury.address,
+        subjectVestingAmount,
+        subjectVestingStart,
+        subjectVestingCliff,
+        subjectVestingEnd,
+      );
+
+      await index.transfer(subjectFtcVesting.address, ether(1000));
+    });
+
+    async function subject(): Promise<ContractTransaction> {
+      return await subjectFtcVesting.connect(treasury.wallet).clawback();
+    }
+
+    it("should clawback funds", async () => {
+      const unclaimedAmount = await index.connect(treasury.wallet).balanceOf(treasury.address);
+      await subject();
+
+      const claimedAmount = await index.connect(treasury.wallet).balanceOf(treasury.address);
+      const hasClaimed = claimedAmount.gt(unclaimedAmount);
```

Comment:
Can you be more specific here? We should ensure that 100% of the funds have been transferred from the vesting contract to the treasury after the clawback rather than just ensuring some amount has been moved. Also, can we use some more descriptive variables for `unclaimedaAmount` and `claimedAmount`. Maybe `initTreasuryBalance` and `finalTreasuryBalance`